src/features/expertConsole/TerminalConsole.tsx

```diff
@@ -227,22 +241,31 @@ function TerminalConsole(props: TConsoleProps): JSX.Element {
 
   const rpcCommandResponse = (commandKey: string) => async (
     state: any,
-    opts: string,
+    opts: string[],
   ) => {
     try {
+      setExecutingCommand(true)
       const data = await rpcApi[commandKey](opts)
       const text =
         typeof data === 'object'
           ? textAsTable(
               Array.isArray(data)
-                ? data.map(i => formatConsoleData(i))
-                : [formatConsoleData(data)],
+                ? data.length > 0
+                  ? data.map(i => formatConsoleData(i))
+                  : ['']
+                : [formatConsoleData(data || {})],
```

Comment:
This is a bit complicated. Can we please break it down into multiple functions or statements?